app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/AdditionalOptions.kt

```diff
@@ -0,0 +1,158 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ *
+ */
+
+package com.wire.android.ui.home.messagecomposer
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material3.Divider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import com.wire.android.R
+import com.wire.android.ui.common.button.WireButtonState
+import com.wire.android.ui.common.button.WireSecondaryIconButton
+import com.wire.android.ui.home.messagecomposer.state.AdditionalOptionMenuState
+import com.wire.android.ui.home.messagecomposer.state.AdditionalOptionStateHolder
+import com.wire.android.ui.home.messagecomposer.state.AdditionalOptionSubMenuState
+import com.wire.android.ui.home.messagecomposer.state.RichTextMarkdown
+import com.wire.android.ui.theme.wireColorScheme
+
+@Composable
+fun AdditionalOptionButton(isSelected: Boolean, isEnabled: Boolean, onClick: () -> Unit) {
+    WireSecondaryIconButton(
+        onButtonClicked = onClick,
+        iconResource = R.drawable.ic_add,
+        contentDescription = R.string.content_description_attachment_item,
+        state = if (!isEnabled) WireButtonState.Disabled
+        else if (isSelected) WireButtonState.Selected else WireButtonState.Default,
+    )
+}
+
+@Composable
+fun AdditionalOptionsMenu(
+    additionalOptionsStateHolder: AdditionalOptionStateHolder,
+    onOnSelfDeletingOptionClicked: (() -> Unit)? = null,
+    onAttachmentOptionClicked: (() -> Unit)? = null,
+    onMentionButtonClicked: (() -> Unit)? = null,
+    onGifOptionClicked: (() -> Unit)? = null,
+    onPingOptionClicked: (() -> Unit)? = null,
+    onRichTextButtonClicked: (RichTextMarkdown) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    var additionalOptionState: AdditionalOptionMenuState by remember { mutableStateOf(AdditionalOptionMenuState.AttachmentAndAdditionalOptionsMenu) }
+
+    Box(modifier) {
+        when (additionalOptionState) {
+            is AdditionalOptionMenuState.AttachmentAndAdditionalOptionsMenu -> {
+                AttachmentAndAdditionalOptionsMenuItems(
+                    isMentionActive = onMentionButtonClicked != null,
+                    isFileSharingEnabled = onAttachmentOptionClicked != null,
+                    onMentionButtonClicked = onMentionButtonClicked ?: {},
+                    onAttachmentOptionClicked = { additionalOptionsStateHolder.toggleAttachmentOptions() },
```

Comment:
It no longer clears focus and closes the keyboard when attachment options are being open which means when the input is focused, attachment options will always be hidden under the keyboard